### PR TITLE
Internal consensus mode of Inner Ring with Sidechain auto-deploy

### PR DESCRIPTION
**Pull request by cthulhu-rider**
pls have a look and advisable pull+try

i'll write commit message finally

### CODE REVIEW

**Comment by vvarg229** [comment]
```
Couldn't find env file: /home/runner/work/neofs-node/neofs-node/neofs-dev-env/services/ir/.ir.env
51eb2f8a8b0e1512ad56694933618460e42c6c9c60140df8de0f66b17fe14a26
Waiting for transfer to settle
fa3fd18095dcd6a08c8cedb21e62fba324b3aee25a7046bf719d1252cd46cde8
Contract: 47b8f6a809c7fe27f2f7e895231a1e437420c[64](https://github.com/vvarg229/neofs-node/actions/runs/7252535413/job/19757223163?pr=50#step:23:65)2
Waiting for deployment to happen
Running bin/deposit.sh
df3ae089fd09aeb2c798c6e9b737009e2388baa8923cd0ea90f169beb921475d
Transfer GAS to wallet ./services/storage/wallet01.json
Error: can't create N3 client: failed to get network magic: Post "http://ir01.neofs.devenv:30333": dial tcp 192.1[68](https://github.com/vvarg229/neofs-node/actions/runs/7252535413/job/19757223163?pr=50#step:23:69).130.61:30333: connect: no route to host
: Failed to transfer GAS to alphabet wallets
make[1]: *** [Makefile:117: up/bootstrap] Error 1
make[1]: Leaving directory '/home/runner/work/neofs-node/neofs-node/neofs-dev-env'
Test environment setup failed. Please check the error messages above.
make: *** [Makefile:252: prepare-test-env] Error 1
```

https://github.com/vvarg229/neofs-node/actions/runs/7252535413/job/19757223163?pr=50

File: services/ir/artifacts.mk [hunk 1]
```diff
@@ -1,34 +1,13 @@
...
-	gzip -dfk ${LOCODE_DB_ARCHIVE_PATH}/${LOCODE_DB_ARCHIVE_FILE}
+get.ir: get.cli get.contracts get.adm get.storage prepare.storage
 
 # Download NeoFS CLI 
 .ONESHELL:
 get.cli: NEOFS_CLI_FILE=./vendor/neofs-cli
 get.cli: NEOFS_CLI_ARCHIVE_FILE=${NEOFS_CLI_FILE}.tar.gz
 get.cli: NEOFS_CLI_PATH?=
 get.cli:
-	@touch services/ir/.ir.env # https://github.com/docker/compose/issues/3560
```
Comment (cthulhu-rider) [inline services/ir/artifacts.mk:31]: @vvarg229 i accidentally removed this line during rebase, seems like https://github.com/nspcc-dev/neofs-dev-env/pull/282#issuecomment-1861384504 caused by this

File: bin/update_hosts.sh [hunk 2]
```diff
@@ -3,28 +3,29 @@ set -e
...
+# Get the NeoFS chain IP address from the $COREFILE file
+neofs_chain_ip=$(grep "ir01.neofs.devenv" "$temp_file" | awk '{print $1}')
 
 # Replace the IP address in the $COREFILE file
-sed -i -E "s/(nns[[:space:]]*http\:\/\/)([0-9]{1,3}[.]){3}[0-9]{1,3}(:30333)/\1$morph_chain_ip\3/" "$COREFILE"
+sed -i -E "s/(nns[[:space:]]*http\:\/\/)([0-9]{1,3}[.]){3}[0-9]{1,3}(:30333)/\1$neofs_chain_ip\3/" "$COREFILE"
 
 # Get the line numbers of "Addresses:"
-addresses_lines=$(grep -n "Addresses:" "$MORPH_CHAIN_CONFIG" | cut -d ':' -f 1)
+# FIXME: grep by 'listen:' is unstable, jump to exact YAML fields
```
Comment (roman-khimov) [inline bin/update_hosts.sh:19]: Likely can be fixed with some `jq` incantation.

**Review by roman-khimov — COMMENTED** [review]
Looks legit, can be 0.39.1 now.

File: bin/update_hosts.sh (hunk 2 above)
Comment (cthulhu-rider) [reply bin/update_hosts.sh:19]: do you mind fixing this separately? the approach was the same before

Comment (vvarg229) [reply bin/update_hosts.sh:19]: > Likely can be fixed with some `jq` incantation.

The jq is sometimes not installed on the custom runners.

**Review by vvarg229 — DISMISSED** [review]

File: bin/update_hosts.sh (hunk 2 above)
Comment (roman-khimov) [reply bin/update_hosts.sh:19]: > do you mind fixing this separately

Issue?

> The jq is sometimes not installed

We already use it in scripts, so I doubt dev-env would work without it.

Comment (cthulhu-rider) [reply bin/update_hosts.sh]: @roman-khimov #302 (and #303)

**Review by vvarg229 — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**